vpr/src/place/place.cpp

```diff
@@ -1375,13 +1516,26 @@ static void update_move_nets(int num_nets_affected) {
     auto& cluster_ctx = g_vpr_ctx.clustering();
     auto& place_move_ctx = g_placer_ctx.mutable_move();
 
+    int num_layers = g_vpr_ctx.device().grid.get_num_layers();
+
     for (int inet_affected = 0; inet_affected < num_nets_affected;
          inet_affected++) {
         ClusterNetId net_id = ts_nets_to_update[inet_affected];
 
-        place_move_ctx.bb_coords[net_id] = ts_bb_coord_new[net_id];
-        if (cluster_ctx.clb_nlist.net_sinks(net_id).size() >= SMALL_NET)
-            place_move_ctx.bb_num_on_edges[net_id] = ts_bb_edge_new[net_id];
+        if (num_layers == 1) {
+            place_move_ctx.bb_coords[net_id] = ts_bb_coord_new[net_id];
+        } else {
+            place_move_ctx.layer_bb_coords[net_id] = layer_ts_bb_coord_new[net_id];
+            place_move_ctx.num_sink_pin_layer[net_id] = ts_layer_sink_pin_count[net_id];
+        }
+
+        if (cluster_ctx.clb_nlist.net_sinks(net_id).size() >= SMALL_NET) {
+            if (num_layers == 1) {
+                place_move_ctx.bb_num_on_edges[net_id] = ts_bb_edge_new[net_id];
+            } else {
+                place_move_ctx.layer_bb_num_on_edges[net_id] = layer_ts_bb_edge_new[net_id];
```

Comment:
It seems you might have done cube_bb in a bit of an odd way, where you compute a bb per layer and then go through and compute an overall bb.  More in common with the 2D case and probably a bit faster to just directly compute one 3D bounding box, which I think the scalar ts_bb* data structures can do.